barretenberg/cpp/src/barretenberg/vm2/common/standard_affine_point.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <cstdint>
+
+namespace bb::avm2 {
+
+template <typename AffinePoint> class StandardAffinePoint {
+  public:
+    using BaseField = AffinePoint::Fq;
+    using ScalarField = AffinePoint::Fr;
+
+    constexpr StandardAffinePoint() noexcept
+        : point(AffinePoint())
+    {}
+
+    constexpr StandardAffinePoint(AffinePoint val) noexcept
+        : point(val)
+    {}
+
+    constexpr StandardAffinePoint(BaseField x, BaseField y, bool is_infinity) noexcept
+    {
+        if (is_infinity) {
+            point = AffinePoint::infinity();
+        } else {
+            point = AffinePoint(x, y);
+        }
```

Comment:
You can do this in the initializer list (I think) and it will avoid 1 default initialization.
```
constexpr StandardAffinePoint(BaseField x, BaseField y, bool is_infinity) noexcept : point(is_infinity ? AffinePoint::infinity() : AffinePoint(x,y)) {}
```